qemu/tests/balloon_service.py

```diff
@@ -21,182 +16,98 @@ def run(test, params, env):
-                session.cmd(start_cmd)
-            except ShellCmdError:
-                raise exceptions.TestError("Start balloon service failed!")
-        else:
-            logging.debug("Install Balloon Service in guest.")
-            try:
-                install_cmd = params["install_balloon_service"] % drive_letter
-                session.cmd(install_cmd)
-            except ShellCmdError:
-                raise exceptions.TestError("Install balloon service failed !")
-
-    def memory_check(vm, get_polling_output, keyname):
-        """
-        Check memory status.
-
-        :param vm: VM object.
-        :param get_polling_output: output of get polling in qmp.
-        :param keyname: key name of the output of the 'qom-get' property.
-        """
-        check_mem_ratio = float(params.get("check_mem_ratio", 0.1))
-
-        error_context.context("Get memory from guest", logging.info)
-        mem_base = MemoryBaseTest(test, params, env)
-        if keyname == "stat-free-memory":
-            guest_mem = mem_base.get_guest_free_mem(vm)
-        elif keyname == "stat-total-memory":
-            guest_mem = mem_base.get_vm_mem(vm)
-
-        error_context.context("Get memory from qmp", logging.info)
-        stat_memory_qmp = get_polling_output['stats'][keyname]
-        stat_memory_qmp = "%sB" % stat_memory_qmp
-        stat_memory_qmp = int(float(utils_misc.normalize_data_size(
-                                   (stat_memory_qmp), order_magnitude="M")))
-
-        error_context.context("Compare memory from guest with qmp",
-                              logging.info)
-        if (abs(guest_mem - stat_memory_qmp) / guest_mem) > check_mem_ratio:
-            raise exceptions.TestFail("%s of guest %s is not equal to %s in"
-                                      " qmp, the acceptable ratio is %s" %
-                                      (keyname, guest_mem, stat_memory_qmp,
-                                       check_mem_ratio))
-
-    def balloon_memory(vm, device_path, mem_check):
-        """
-        Doing memory balloon in a loop and check memory status during balloon.
+        Doing memory balloon in a loop and check memory statistics during balloon.
 
         :param vm: VM object.
         :param device_path: balloon polling path.
         :param mem_check: need to do memory check if param mem_check is 'yes'
         """
         repeat_times = int(params.get("repeat_times", 5))
         logging.info("repeat times: %d" % repeat_times)
+        base_path = params.get("base_path", "/machine/peripheral/")
+        device = params.get("balloon", "balloon0")
+        device_path = os.path.join(base_path, device)
 
-        if params['os_type'] == 'windows':
-            balloon_test = BallooningTestWin(test, params, env)
-        else:
-            balloon_test = BallooningTestLinux(test, params, env)
         min_sz, max_sz = balloon_test.get_memory_boundary()
```

Comment:
I suggest to move this line out of the function balloon_memory(). We don't need to get the boundary every time but only at the beginning.

Reply:
Will update it.

Thanks